### ISSUE TITLE
colors are looking weird on cmp menu

### ISSUE THREAD

**Issue by datsfilipe** [nyoom-engineering/oxocarbon.nvim#45]
Look, the "f"s from function items are with the background colored, not the text. It changes when they're in selection line:
![image](https://user-images.githubusercontent.com/76636791/216798086-18c3a5c2-4e90-41b0-962d-00d7813007dc.png)

Also, It can be a thing of mine configuration, but seemed to be that there's a little lag before the right colors are applied. Maybe it's something related to treesitter. The colors do not apply fast as other colorschemes, they take like 1-2 seconds to apply.

**Comment by datsfilipe** [comment]
Didn't experienced the lag again, this might be something on other plugins too and the cmp menu seems that has been planned to be that way (I did not paid attention, sry), so there's everything okay.

**Comment by fitrh** [comment]
Yes, the `CmpItemKind*` highlight sets the background color, I think it's due to
this theme is very nyoom.nvim oriented, for me, I did something like this to
overrides `CmpItemKind*` highlight

```lua
for _, kind in ipairs({
  "Interface",
  "Color",
  "TypeParameter",
  "Text",
  "Enum",
  "Keyword",
  "Constant",
  "Constructor",
  "Reference",
  "Function",
  "Struct",
  "Class",
  "Module",
  "Operator",
  "Field",
  "Property",
  "Event",
  "Unit",
  "Snippet",
  "Folder",
  "Variable",
  "File",
  "Method",
  "Value",
  "EnumMember",
}) do
  local group = string.format("CmpItemKind%s", kind)
  local bg = vim.api.nvim_get_hl_by_name(group, true)['background']
  bg = bg and string.format('#%06X', bg) or 'NONE'

  vim.api.nvim_set_hl(0, group, { fg = bg })
end

```

**Comment by shaunsingh** [comment]
Indeed the theme is pretty much directly copied from nyoom with a small patch, I will add an option to use the "regular" cmp color for non-nyoom users

**Comment by boydkelly** [comment]
> Look, the "f"s from function items are with the background colored, not the text. It changes when they're in selection line: ![image](https://user-images.githubusercontent.com/76636791/216798086-18c3a5c2-4e90-41b0-962d-00d7813007dc.png)
> 
> Also, It can be a thing of mine configuration, but seemed to be that there's a little lag before the right colors are applied. Maybe it's something related to treesitter. The colors do not apply fast as other colorschemes, they take like 1-2 seconds to apply.

Awesome theme.   Would really appreciate if the completion menu can be fixed.  Thx!!!

**Comment by boydkelly** [comment]
I can offer a pull request for this.   I find that making the bg the same as the menu or setting it to oxocarbon.none has the same effect.   Is there any preference?

**Comment by boydkelly** [comment]
> Yes, the `CmpItemKind*` highlight sets the background color, I think it's due to this theme is very nyoom.nvim oriented, for me, I did something like this to overrides `CmpItemKind*` highlight

Also another possiblilty is to make the completion menu like nyoom.  Just add 

````
        formatting = {
          fields = { "kind", "abbr", "menu" },
````

to the cmp.setup and it will put the symbol at the begining of the line of the completion menu.  Its starting to look better, but I just need to figure out how to add some space around the symbol....

![Screenshot from 2023-03-23 16-02-08](https://user-images.githubusercontent.com/3412972/227263545-9ed5aeaa-353d-41e3-b84b-717781d320b2.png)

**Comment by boydkelly** [comment]
Ok I got it.   In addition to cmp.setup formatting fields,   you need to add (with spaces surrounding the symbols)

````
 require("lspkind").init({
 symbol_map = {
 Text = "  ",
````
![Screenshot from 2023-03-23 16-31-22](https://user-images.githubusercontent.com/3412972/227271643-59cf6b84-e528-42cd-8888-9565fd63aaa2.png)

**Comment by datsfilipe** [comment]
@boydkelly it looks prettier but what I was complaining about was mainly because it is coloring the background instead of the symbol like other themes often do. Also, the main issue has become the fact that the theme is very 'Nyoom.nvim' oriented and not very independent I think. At least that's what is being said here:

> Or remove all nyoom.nvim-specific stuff from this theme and let nyoom.nvim configure this theme

*Note: I'm not using the theme right now, I'm kind of waiting for changes since I like it, but thanks anyway for trying to help!*

**Comment by yousef8** [comment]
> @boydkelly 

> Ok I got it. @datsfilipe In addition to cmp.setup formatting fields, you need to add (with spaces surrounding the symbols)
> 
> ```
>  require("lspkind").init({
>  symbol_map = {
>  Text = "  ",
> ```
> 
> ![Screenshot from 2023-03-23 16-31-22](https://user-images.githubusercontent.com/3412972/227271643-59cf6b84-e528-42cd-8888-9565fd63aaa2.png)

I had this issue and was very sad because I very like the theme but your solution solved my issue thank you very much. I really hope you fix this in the colorscheme it self in the near future

**Comment by avikrao** [comment]
+1 still waiting on this, please make it possible to set icon (foreground) color instead of background color

**Comment by becknik** [comment]
> Yes, the `CmpItemKind*` highlight sets the background color, I think it's due to this theme is very nyoom.nvim oriented, for me, I did something like this to overrides `CmpItemKind*` highlight
> 
> for _, kind in ipairs({
>   "Interface",
>   "Color",
>   "TypeParameter",
>   "Text",
>   "Enum",
>   "Keyword",
>   "Constant",
>   "Constructor",
>   "Reference",
>   "Function",
>   "Struct",
>   "Class",
>   "Module",
>   "Operator",
>   "Field",
>   "Property",
>   "Event",
>   "Unit",
>   "Snippet",
>   "Folder",
>   "Variable",
>   "File",
>   "Method",
>   "Value",
>   "EnumMember",
> }) do
>   local group = string.format("CmpItemKind%s", kind)
>   local bg = vim.api.nvim_get_hl_by_name(group, true)["background"]
>   bg = bg and string.format("#%06X", bg) or "NONE"
> 
>   vim.api.nvim_set_hl(0, group, { fg = bg })
> end

Thanks for your idea, I adjusted it since `nvim_get_hl_by_name` is deprecated as of 0.9 & to make it a bit shorter:

```lua
for i, kind in ipairs(vim.lsp.protocol.CompletionItemKind) do
  local group = string.format("CmpItemKind%s", kind)
  local bg = vim.api.nvim_get_hl(0, { name = group })["bg"]
  bg = bg and string.format("#%06X", bg) or "NONE"
  vim.api.nvim_set_hl(0, group, { fg = bg })
end
```
